### PR TITLE
Migrate guides for 1.12 through 1.14 to ones for 1.15

### PR DESCRIPTION
**Pull request by tnir**
### What was the end-user problem that led to this PR?

Guides for between 1.12 and 1.15 are duplicated.

### What was your diagnosis of the problem?

Ones for 1.12 are still compatible with ones for 1.15 as confirmed in #668.

### What is your fix for the problem, implemented in this PR?

Guides for between 1.12 and 1.14 are removed and redirected to ones for 1.15.

#### Removal

- `(cd source/v1.15/guides; git ls-files | sed 's/.html.*//') | xargs -I {} git rm source/v{1.12,1.13,1.14}/{}.html.*`
  - files for guides for v1.14 and below are not structured since they were done before #322.
- `git rm source/v1.12/rails*.html.*`
  - v1.12 has `rails23` and `rails3` instead of `rails`.

#### Redirection

See changes in `config.rb`.

### Why did you choose this fix out of the possible options?

With this change, `/v1.12/rails.html` (with redirection to `/v1.15/rails.html`) would be generated, but gave up for simplicity.

Signed-off-by: Takuya Noguchi [takninnovationresearch@gmail.com](https://github.com/sponsors/tnir)

### CODE REVIEW

**Comment by duckinator** [comment]
@tnir is this good to go on your end?

**Comment by tnir** [comment]
@duckinator I do not have anything from my side. Just rebased the branch to trigger a pipeline to deploy this to Heroku, but does not work 😢 and you can start to review this 🚀

**Comment by deivid-rodriguez** [comment]
@tnir I sent you an invite so that you have push access to the repo. If you accept the invitation and create further PRs from a branch inside the repository, review apps should get created normally 👍.

**Comment by deivid-rodriguez** [comment]
For this PR I created the review app manually.

**Review by deivid-rodriguez — APPROVED** [review]
Looks good, I'll approve a let @duckinator merge if she's ok!

**Review by duckinator — APPROVED** [review]

File: config.rb [hunk 1]
```diff
@@ -85,13 +85,20 @@
...
+  end
+
+  # Redirect old localizable guides (v1.12-v1.14) to the latest version (1.15) of guide compatible with Ruby 1.8.x
+  ["", "pl/"].each do |lang|
     redirect "#{lang}v#{version}/guides/creating_gem.html", to: "#{lang}v1.15/guides/creating_gem.html"
     redirect "#{lang}v#{version}/guides/using_bundler_in_applications.html", to: "#{lang}v1.15/guides/using_bundler_in_applications.html"
   end
 end
+%w[rails23 rails3].each do |filename|
+  redirect "v1.12/guides/#{filename}.html", to: "v1.15/guides/#{filename}.html"
```
Comment (tnir) [inline config.rb:100]: This was wrong and is now fixed in #1027 🙇

**Review by tnir — COMMENTED** [review]
😭 😭

**Pull request merged**